src/XCCDF_POLICY/xccdf_policy.c

```diff
@@ -2253,6 +2281,44 @@ void xccdf_policy_model_free(struct xccdf_policy_model * model) {
         free(model);
 }
 
+static const char *_find_reference_uri_by_key(struct xccdf_benchmark *benchmark, const char *key)
+{
+	const char *uri = NULL;
+	struct oscap_reference_iterator *benchmark_references = xccdf_item_get_references((struct xccdf_item *)benchmark);
+	while (oscap_reference_iterator_has_more(benchmark_references)) {
+		struct oscap_reference *ref = oscap_reference_iterator_next(benchmark_references);
+		const char *title = oscap_reference_get_title(ref);
+		if (!strcmp(key, title)) {
+			uri = oscap_reference_get_href(ref);
+			break;
+		}
+	}
+	oscap_reference_iterator_free(benchmark_references);
+	return uri;
+}
+
+void xccdf_policy_set_reference_filter(struct xccdf_policy *policy, const char *reference_parameter)
+{
+	if (!reference_parameter) {
+		return;
+	}
+	char *reference_parameter_dup = strdup(reference_parameter);
+	char **split = oscap_split(reference_parameter_dup, ":");
```

Comment:
So I can get a segfault if I run `./oscap_wrapper xccdf eval --profile stig --reference 'stigid'  [datastream]`. There should be some error handling for this case.